ts/sagas/startup/checkAcceptedTosSaga.ts

```diff
@@ -1,30 +1,25 @@
 import { Effect } from "redux-saga";
-import { put, select, take } from "redux-saga/effects";
-import { getType } from "typesafe-actions";
+import { put, take } from "redux-saga/effects";
 
+import { UserProfileUnion } from "../../api/backend";
+import { tosVersion } from "../../config";
 import { navigateToTosScreen } from "../../store/actions/navigation";
-import { tosAccept } from "../../store/actions/onboarding";
-import { isTosAcceptedSelector } from "../../store/reducers/onboarding";
-import { GlobalState } from "../../store/reducers/types";
+import { profileUpsert } from "../../store/actions/profile";
 
-export function* checkAcceptedTosSaga(): IterableIterator<Effect> {
-  // From the state we check whether the user has already accepted the ToS
-  // FIXME: ToS can change over time, this step should eventually check whether
-  //        the user has accepted the latest version of the ToS and store the
-  //        information in the user profile.
-  const isTosAccepted: ReturnType<typeof isTosAcceptedSelector> = yield select<
-    GlobalState
-  >(isTosAcceptedSelector);
-
-  if (!isTosAccepted) {
-    // Navigate to the TosScreen
-    yield put(navigateToTosScreen);
+export function* checkAcceptedTosSaga(
+  userProfile: UserProfileUnion
+): IterableIterator<Effect> {
+  if (
```

Comment:
```suggestion
  if (
    profileState.value.has_profile &&
    profileState.value.accepted_tos_version === tosVersion
```

Reply:
Isn't this better?

Reply:
I don't include it having on the main saga some controls on the existence of the profile (such as the way userProfile is declared - startup.ts saga, lines from 162 to 167). Aren't they enough to guarantee the profile exisists?

Reply:
The has_profile flag is use differently ([here](https://github.com/teamdigitale/io-app/pull/1047/files#diff-f02ab1035adcfaf43c32bff478e170e0R30) to avoid the upsert request occurs also for not yet unboarded user) while placed here it would deny the user access the TosScreen

Reply:
has_profile is a flag added by the backend when the user has a profile API side.